### PR TITLE
don't autoexecute query on changing datasource

### LINKED ISSUE #2651: Don't execute auto-execute query when changing datasources
Copies https://github.com/mozilla/redash/issues/436 to port https://github.com/mozilla/redash/issues/29 upstream.

Right now when you change datasource the query automatically executes. This is an issue to delete the automatic execution.

### PR DESCRIPTION
**Pull request by alison985**
fixes #2651 

This is a port of a Mozilla feature upstream.

### CODE REVIEW

**Comment by alison985** [comment]
That makes sense to me Arik. I'm assuming you mean just hide the display of the results from the UI. 

There is already a `ng-if="showDataset"` in the query.html page for the tabs and data of query results. I tried adding `$scope.showDataset = false;` locally and adding a setter for `showDataset` to fix this error: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Getter_only but that didn't work so I added a new variable.

**Comment by arikfr** [comment]
> That makes sense to me Arik. I'm assuming you mean just hide the display of the results from the UI.

No, actually the important thing is to clear the reference to the last result (`latest_query_data_id`). Because otherwise someone loading this query will see that it uses some data source that the results don't belong to.

**Comment by alison985** [comment]
@arikfr That's already happening here: https://github.com/getredash/redash/blob/master/client/app/pages/queries/view.js#L305

**Comment by alison985** [comment]
@arikfr Do I need to do anything else before this can be merged?

**Comment by jezdez** [comment]
@alison985 Let's wait till Arik is back for this to merge.

File: client/app/pages/queries/view.js [hunk 1]
```diff
@@ -120,6 +120,7 @@ function QueryViewCtrl(
     Notifications.getPermissions();
   };
 
+  $scope.dataSourceHasNotChanged = true;
```
Comment (jezdez) [inline client/app/pages/queries/view.js]: Let's turn around the logic and call this `dataSourceChanged` with a default of `false` so we don't have to deal with a double negative below.

Comment (alison985) [reply client/app/pages/queries/view.js]: @jezdez Updated.

Comment (jezdez) [reply client/app/pages/queries/view.js]: @alison985 Thank you! 😃

**Comment by alison985** [comment]
@jezdez Ready for Review.

File: client/app/pages/queries/query.html [hunk 2]
```diff
@@ -217,7 +217,9 @@ <h3>
               <!-- End of Query Execution Status -->
 
               <!-- tabs and data -->
-              <div ng-if="showDataset" class="flex-fill p-relative">
+              <div ng-if="showDataset && dataSourceChanged" class="flex-fill p-relative">
+                <div class="d-flex flex-column p-absolute" style="left: 0; top: 0; right: 0; bottom: 0;">
```
Comment (jezdez) [inline client/app/pages/queries/query.html]: This opens a new `div` without closing it, and also uses inline styles. Is this needed or just for debugging purposes?

Comment (alison985) [reply client/app/pages/queries/query.html]: These lines actually look like the result of a bad merge as they serve no purpose. I've removed them. The in-line styles below were there before me.

File: client/app/pages/queries/query.html [hunk 3]
```diff
@@ -217,7 +217,7 @@ <h3>
               <!-- End of Query Execution Status -->
 
               <!-- tabs and data -->
-              <div ng-if="showDataset" class="flex-fill p-relative">
+              <div ng-if="showDataset && dataSourceChanged" class="flex-fill p-relative">
```
[the remark this answers is not in the document]
Comment (alison985) [reply client/app/pages/queries/query.html]: @arikfr This variable was originally `dataSourceHasNotChanged` but @jezdez wanted to remove the double negative. This is the variable added to handle your comments above to not show results when the datasource has changed. You're right to question this, I will add the ! now.

**Comment by alison985** [comment]
Ping @jezdez and @arikfr

**Comment by alison985** [comment]
Docker Hub issue made build fail. When Docker Hub is back up the build should be rerun.

**Comment by alison985** [comment]
Anything else on this one @jezdez and/or @arikfr?

**Comment by chang** [comment]
+1 This could be especially dangerous when developing a mutation query on a test database. If the schemas are similar enough, it could successfully and unintentionally auto-execute while switching over to a prod data source.

**Comment by arikfr** [comment]
Hi,

(This is a template message, but I mean every word of it. Also you're welcome to reply)

Thank you for making this contribution. While we couldn't bring it to completion and merge, it's still very much appreciated. 🙇

In the past year the Redash code base gone under massive updates: on the backend we moved to Python 3 & RQ instead of Celery and on the frontend we replaced Angular with React. It's very likely this makes this PR irrelevant without significant changes. :-( 

I'm closing this PR now. But if you're still interested in making it happen, let me know and I will reopen.

Thanks.

**Comment by jezdez** [comment]
@arikfr I'm a bit confused by your last review comment, you're right that changing the source is only possible in edit mode, but I still see a request happening against the server when the source is changed there, and I believe the query is actually executed. I'd appreciate if you'd double-check this on your side so we can close this.

Here's a screen recording of the current master executing the query when changing a data source (if it's the same type of data source): https://dsc.cloud/a46d84/RhVthKiRVrg1Od79fBzY

**Comment by arikfr** [comment]
I just tried it out on https://redash-preview.netlify.com/ and couldn't reproduce it. And looking at your video again I can see that it's definitely not the latest master. Maybe you didn't rebuild the assets?

**Comment by jezdez** [comment]
Lemme check again, thanks!

**Comment by jezdez** [comment]
You right, @arikfr, that was it :) Sorry for the noise!

**Pull request closed without merging**